middleware/nethttp/handler.go

```diff
@@ -0,0 +1,50 @@
+package wasm
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/api"
+)
+
+const (
+	FuncHandle = "handle"
+	Memory     = "memory"
+)
+
+// Handler is a http.Handler implemented by a Wasm module. `ServeHTTP` is
+// dispatched to FuncHandle, which is exported by the guest.
+type Handler interface {
+	http.Handler
+
+	// Close releases the Wasm module associated with this handler.
+	Close(context.Context) error
+}
+
+// compile-time check to ensure wasmHandler implements Handler.
+var _ Handler = &wasmHandler{}
+
+type wasmHandler struct {
+	ns    wazero.Namespace
+	guest api.Module
+	next  http.Handler
+}
+
+// ServeHTTP implements http.Handler
+func (w *wasmHandler) ServeHTTP(response http.ResponseWriter, request *http.Request) {
+	// The guest Wasm actually handles the request. As it may call host
+	// functions, we add context parameters of the current request.
+	ctx := withRequestState(request.Context(), response, request, w.next)
+	if _, err := w.guest.ExportedFunction("handle").Call(ctx); err != nil {
+		// TODO: after testing, shouldn't send errors into the HTTP response.
+		response.Write([]byte(err.Error())) // nolint
```

Comment:
Not sure we want to expose the error and leak implementation details. Can't we drop a log instead?

Reply:
this is a sketch, so that's exactly why the TODO is there. Ironically, I had hoped that by commenting we could avoid jumping into weeds like this and instead focus on the wasm part ;)

Reply:
I'll re-clarify the comment that this line is intentional until more progress is made and we can decide a logging strategy. I should have earlier linked back to this https://github.com/http-wasm/http-wasm-abi/pull/2#issuecomment-1261752423

basically the idea is to try to not get swamped in details of little code features as the main thing is getting infrastructure in place. I have confidence things like noodling on debug log strategy can be addressed, but it is also not the right time as there aren't even multiple impls yet.